### ISSUE TITLE
Accel-ppp IPoE incompatibility with kernel 6.1

### ISSUE THREAD

**Issue by c-po** [accel-ppp/accel-ppp#74]
Compiling Accel-PPP for Linux Kernel 6.1.7 results in the vlan_mon and ipoe Kernel modules to no longer function.

Initially reported here: https://phabricator.vyos.net/T4941

```
Jan 18 14:22:29 VyOS-BSDRP-Router accel-ipoe: cli: tcp: new connection from 127.0.0.1
Jan 18 14:25:03 VyOS-BSDRP-Router accel-ipoe: cli: tcp: new connection from 127.0.0.1
Jan 18 14:25:03 VyOS-BSDRP-Router accel-ipoe: terminate, sig = 15
Jan 18 14:25:03 VyOS-BSDRP-Router accel-ipoe: ipoe: stop interface eth4.50
Jan 18 14:25:03 VyOS-BSDRP-Router accel-ipoe: ipoe: nl_del_iface: Invalid argument
Jan 18 14:25:03 VyOS-BSDRP-Router accel-ipoe: ipoe: stop interface eth4.51
Jan 18 14:25:04 VyOS-BSDRP-Router accel-ipoe: ipoe: start interface eth4.50 (shared=1,mode=L2,ifcfg=1,range=172.16.50.0/24,start=dhcpv4,ipv6=1)
Jan 18 14:25:04 VyOS-BSDRP-Router accel-ipoe: ipoe: start interface eth4.51 (shared=1,mode=L2,ifcfg=1,range=172.16.98.0/24,start=dhcpv4,ipv6=1)
Jan 18 14:25:04 VyOS-BSDRP-Router accel-ipoe: vlan_mon: nl_del_vlan_mon: error talking to kernel
Jan 18 14:27:27 VyOS-BSDRP-Router accel-ipoe: send [DHCPv4 Nak xid=14f5a1bd chaddr=00:0c:29:b7:49:a7 <Message-Type Nak> <Server-ID 255.255.255.255> <Message Session dosn't exist>]
Jan 18 14:27:27 VyOS-BSDRP-Router accel-ipoe: send [DHCPv4 Nak xid=20779fb8 chaddr=00:0c:29:b7:49:a7 <Message-Type Nak> <Server-ID 255.255.255.255> <Message Session dosn't exist>]
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: eth4.51:: recv [DHCPv4 Discover xid=577891e8 chaddr=00:0c:29:b7:49:a7 <Message-Type Discover> <Host-Name VyOS-PPPoE-IPoE-Client> <Request-List Subnet,Broadcast,Router,DNS,Classless-Route,Domain-Name,MTU>]
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: ipoe0:: create interface ipoe0 parent eth4.51
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: ipoe0:eth4.51: eth4.51: authentication succeeded
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: ipoe0:eth4.51: send [DHCPv4 Offer xid=577891e8 yiaddr=172.16.98.2 chaddr=00:0c:29:b7:49:a7 <Message-Type Offer> <Server-ID 172.16.98.1> <Lease-Time 600> <T1 300> <T2 525> <Router 172.16.98.1> <Subnet 255.255.255.0> <DNS 10.17.6.120,8.8.8.8>]
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: ipoe0:eth4.51: recv [DHCPv4 Request xid=577891e8 chaddr=00:0c:29:b7:49:a7 <Message-Type Request> <Server-ID 172.16.98.1> <Request-IP 172.16.98.2> <Host-Name VyOS-PPPoE-IPoE-Client> <Request-List Subnet,Broadcast,Router,DNS,Classless-Route,Domain-Name,MTU>]
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: ipoe0:eth4.51: ipoe: nl_modify: Invalid argument
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: ipoe0:eth4.51: ipoe: session finished
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: ipoe0:eth4.51: ipoe: nl_delete: Invalid argument
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: eth4.50:: recv [DHCPv4 Discover xid=5abe18e2 chaddr=00:0c:29:b7:49:a7 <Message-Type Discover> <Host-Name VyOS-PPPoE-IPoE-Client> <Request-List Subnet,Broadcast,Router,DNS,Classless-Route,Domain-Name,MTU>]
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: eth4.50:: ipoe: nl_create: Invalid argument
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: eth4.50:: ipoe: failed to create interface
Jan 18 14:27:29 VyOS-BSDRP-Router accel-ipoe: eth4.50:: ipoe: session finished
```

**Comment by micron10** [comment]
Try this patch 
[accelppp-fix-vlan-mon-kernel-6.1.patch](https://github.com/accel-ppp/accel-ppp/files/10475593/accelppp-fix-vlan-mon-kernel-6.1.patch)

i think will fix your problem

**Comment by DmitriyEshenko** [comment]
Should be fixed for ipoe and vlan_mon b120b0d83e2145d1681f10003b13274dec6dc412

**Comment by c-po** [comment]
Thanks Dmitry!
